packages/astro/src/core/request.ts

```diff
@@ -76,8 +76,13 @@ export function createRequest({
 
 	if (staticLike) {
 		// Warn when accessing headers in SSG mode
-		const _headers = request.headers;
+		let _headers = request.headers;
 		const headersDesc = Object.getOwnPropertyDescriptor(request, 'headers') || {};
+
+		// We need to delete any existing descriptor's value and writable properties because we're adding getters and setters.
+		delete headersDesc.value;
+		delete headersDesc.writable;
```

Comment:
Could we do something like
```suggestion
		const { ...headersDesc, value, writable } = Object.getOwnPropertyDescriptor(request, 'headers') || {};
```

Or alternatively only pass specific properties in the `defineProperty` call below? Using `delete` is usually more expensive than either of the two approaches above, so would be nice to avoid that if possible.

Reply:
Agreed!